thoth/solver/python/instrument.py

```diff
@@ -61,7 +61,19 @@ def _get_importlib_metadata_metadata(package_name):
     import importlib_metadata
     import json
 
-    print(json.dumps(dict(importlib_metadata.metadata(package_name).items())))
+    result = dict(importlib_metadata.metadata(package_name).items())
+
+    # As metadata are encoded as email.message, it's not possible to implicitly expose information about which keys
+    # keep multiple values and which are single-value. Let's explicitly maintain a list for metadata keys ara arrays.
+    #    https://setuptools.readthedocs.io/en/latest/setuptools.html#metadata
+    #
+    for key in frozenset(("Classifier", "Requires-Dist", "requires", "provides", "obsoletes", "Provides-Extra")):
```

Comment:
I think "requires", "provides", "obsoletes" are not in `importlib_metadata.metadata`, instead there should be 'Requires-External`, `Provides-Dist`, `Obsoletes-Dist` according to https://packaging.python.org/specifications/core-metadata/#id28 which is taken from last version of importlib_metadata.

Reply:
Thanks for looking this up! I used the old setuptools docs, this is better :+1: